src/analyzers/concept/vehicle-purchase/VehiclePurchaseSolution.ts

```diff
@@ -0,0 +1,213 @@
+import {
+  AstParser,
+  ExtractedFunction,
+  extractExports,
+  extractFunctions,
+  findFirstOfType,
+} from '@exercism/static-analysis'
+import { AST_NODE_TYPES, TSESTree } from '@typescript-eslint/typescript-estree'
+import { readFileSync } from 'fs'
+import path from 'path'
+import { Source } from '~src/analyzers/SourceImpl'
+import { assertPublicApi } from '~src/asserts/assert_public_api'
+import { PublicApi } from '../../PublicApi'
+
+export const NEEDS_LICENSE = 'needsLicense'
+export const CHOOSE_VEHICLE = 'chooseVehicle'
+export const CALCULATE_RESELL_PRICE = 'calculateResellPrice'
+
+/**
+ * Verify that `needsLicense` does not include an unnecessary if-statement where
+ * the student returns `true`/`false`.
+ */
+class NeedsLicense extends PublicApi {
+  constructor(public readonly implementation: ExtractedFunction) {
+    super(implementation)
+  }
+
+  public get isOptimal(): boolean {
+    if (this.implementation.params.length !== 1) {
+      return false
+    }
+
+    let foundSuboptimalNode = false
+
+    this.traverse({
+      enter() {
+        foundSuboptimalNode = true
+      },
+
+      [AST_NODE_TYPES.ReturnStatement]() {
+        foundSuboptimalNode = false
+      },
+
+      [AST_NODE_TYPES.BlockStatement]() {
+        foundSuboptimalNode = false
+      },
+
+      exit() {
+        if (foundSuboptimalNode) {
+          this.break()
+        }
+      },
+    })
+
+    return foundSuboptimalNode
+  }
+
+  public get hasConditional(): boolean {
+    return !!findFirstOfType(
+      this.implementation.body,
+      AST_NODE_TYPES.IfStatement
+    )
+  }
+}
+
+/**
+ * Verify that in `chooseVehicle` the string `' is clearly the better choice'`
+ * only appears once.
+ * Verify the student actually practiced if/else and did not use early returns.
+ */
+class ChooseVehicle extends PublicApi {
+  constructor(public readonly implementation: ExtractedFunction) {
+    super(implementation)
+  }
+
+  public get isOptimal(): boolean {
+    if (this.implementation.params.length !== 2) {
+      return false
+    }
+
+    let foundSuboptimalNode = false
+
+    this.traverse({
+      enter() {
+        foundSuboptimalNode = true
+      },
+
+      [AST_NODE_TYPES.ReturnStatement]() {
+        foundSuboptimalNode = false
+      },
+
+      [AST_NODE_TYPES.BlockStatement]() {
+        foundSuboptimalNode = false
+      },
+
+      exit() {
+        if (foundSuboptimalNode) {
+          this.break()
+        }
+      },
+    })
+
+    return foundSuboptimalNode
+  }
+
+  public get usesIfElse(): boolean | undefined {
+    const body = this.implementation.body
+    if (body.type === AST_NODE_TYPES.BlockStatement) {
```

Comment:
Can you explicitly return `undefined` as an early return? That makes it much more discoverable:

```suggestion
    // Only block statements can be inspected. Use `undefined` to signal
    // the analyzer does not know if it uses if / else.
    if (body.type !== AST_NODE_TYPES.BlockStatement) {
      return undefined;
    }
```